app/src/sharedTest/java/org/oppia/android/app/home/HomeActivityTest.kt

```diff
@@ -1664,7 +1652,7 @@ class HomeActivityTest {
       verifyExactTextOnHomeListItemAtPosition(
         itemPosition = 0,
         targetViewId = R.id.welcome_text_view,
-        stringToMatch = "صباح الخير"
+        stringToMatch = "صباح الخير \u200F\u202AAdmin\u202C\u200F!"
```

Comment:
Could you explain why this has utf space before and after? I think this is why your test is failing, you just need to add a comma (,) and "Admin" to the existing text, and bear in mind that the text is RTL.

Reply:
I am able to pass this test using only utf. I tried with the string "Admin ,مساء الخير", didn't work.

Reply:
Does it work if you revert the change:

  `@RunOn(TestPlatform.ROBOLECTRIC) // TODO(#3840): Make this test work on Espresso & Robolectric.`

It would be nice to explain in your PR description why you removed that line.

Reply:
@RunOn(TestPlatform.ROBOLECTRIC) // TODO(#3840): Make this test work on Espresso & Robolectric,
 makes the test fail in GitHub-action

Reply:
In [this doc,](https://developer.android.com/guide/topics/resources/pseudolocales), it appears that the order of the characters should be reversed. I think if you spell admin in mirror, it should work. Please let me know.

Reply:
Hi @ShubhadeepKarmakar, I just wanted to update that its ok to have the  utf encoding, but you still need to address the other comment about the TODO.

Reply:
I reverted the line, "@RunOn(TestPlatform.ROBOLECTRIC) // TODO(#3840): Make this test work on Espresso & Robolectric."